### PR TITLE
Create log file w abs path so tests pass on MacOS

### PR DESCRIPTION
**Pull request by jhtimmins**
Tests that create log files using a relative path can fail when the due to limited permissions in the current directory, regardless of permissions in the log file directory. By creating log files with an absolute path rather than a relative path, this problem is avoided.

---
Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Target Github ISSUE in description if exists
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

**Review by ashb — COMMENTED** [review]
Should we apply the same expansion to the check above?

File: airflow/utils/log/file_processor_handler.py [hunk 1]
```diff
@@ -144,6 +144,6 @@ def _init_file(self, filename):
                     raise
 
         if not os.path.exists(full_path):
-            open(full_path, "a").close()
+            open(os.path.abspath(full_path), "a").close()
```
Comment (kaxil) [inline airflow/utils/log/file_processor_handler.py]: Do we need this here though??

https://github.com/apache/airflow/blob/05a195c11b1a75c193ae172263642fe80f40025f/airflow/utils/log/file_processor_handler.py#L136

https://github.com/apache/airflow/blob/05a195c11b1a75c193ae172263642fe80f40025f/airflow/utils/log/file_processor_handler.py#L95-L98

https://github.com/apache/airflow/blob/d590e5e7679322bebb1472fa8c7ec6d183e4154a/airflow/config_templates/airflow_local_settings.py#L84

https://github.com/apache/airflow/blob/d590e5e7679322bebb1472fa8c7ec6d183e4154a/airflow/config_templates/airflow_local_settings.py#L49

https://github.com/apache/airflow/blob/d590e5e7679322bebb1472fa8c7ec6d183e4154a/airflow/config_templates/default_airflow.cfg#L692

Comment (jhtimmins) [reply airflow/utils/log/file_processor_handler.py]: @kaxil Not sure I follow. The clients of this method expect to receive the relative path.

Comment (ashb) [reply airflow/utils/log/file_processor_handler.py]: `child_process_log_directory` config setting, should, normally be an absolute path.

Perhaps there should be a validation/transform step (at config load time) that sets it to an absolute path if it isn't?

Comment (ashb) [reply airflow/utils/log/file_processor_handler.py]: Yeah, cos a variable called "full_path" sure sounds like it was expected to be an absolute path already.

**Comment by jhtimmins** [comment]
@ash @kaxil k I updated it so it only ever uses absolute paths

**Comment by jhtimmins** [comment]
I added the `config.env_vars()` for the following reasons:
1. Config vals won't be maintained across processes when created with `fork`. Env vars will.
2. `conf_vars` can't trivially set environment variables (I tried). This caused issues.
3. Arguably, too much logic already lives in `conf_vars`.
4. The current method of modifying env vars directly risks adding too much global state. While this adds some global state, it minimizes the potential impact of said state.

**Comment by kaxil** [comment]
Some of tests timed out it seems

**Review by ashb — APPROVED** [review]

**Comment by ashb** [comment]
One failure:

```
2020-05-14T21:36:29.9829995Z FAILED tests/utils/test_serve_logs.py::TestServeLogs::test_should_serve_file
```

Perhaps that should be set up to retry, or moved to quarantine. Merging this now.

**Pull request merged**